api/medicine/views.py

```diff
@@ -112,28 +112,48 @@ def get(self, request, pk, *args, **kwargs):
             return Response({"error": "no need to take it"}, status=status.HTTP_400_BAD_REQUEST)
         return Response(serializer.data)
 
-
+def get_type_of_user(data):
+    try:
+        # self.request.user.patient
+        data.guardian
+    except Guardian.DoesNotExist as e:
+        try:
+            # self.request.user.patient
+            data.patient
+        except Patient.DoesNotExist as e:
+            return "not found"
+        return "patient"
+    return "guardian"
```

Comment:
Rewrite this method by analogy of whoiam

Reply:
noy